File: Examples/cosmology/ng.py
## ==========================================
## Example code: how to estimate the skewness
## ==========================================

# to ensure compatibility in Python versions 2.x and 3.x
from __future__ import print_function

# import the CosmoBolognaLib module
import CosmoBolognaLib as cbl

# set the cosmological model, using default parameters
cosmology = cbl.LCDM("Planck18")

# create the object used for the computation of the skewness
ng = cbl.NG(cosmology)

print("The skewness at M=1e13 Msun/h is:", ng.skewness(1.e13, "EisensteinHu"))

File: Examples/cosmology/ng.cpp
// ==========================================
// Example code: how to estimate the skewness
// ==========================================

#include "LCDM.h"
#include "NG.h"

using namespace std;

int main () {
    
  try {
    
    // --------------------------------------------------------
    // ---------------- set the cosmological model ------------
    // --------------------------------------------------------
     
    auto cosmology = make_shared<cbl::cosmology::LCDM>("Planck18");

    
    // ---------------------------------------------------
    // ---------------- estimate the skewness ------------
    // ---------------------------------------------------

    cbl::cosmology::NG ng(cosmology);
    
    const double SK = ng.skewness(1.e13, "EisensteinHu");
 
    cout << "The skewness at M=1e13 Msun/h is:" << SK << endl << endl;
    
  }

  catch(cbl::glob::Exception &exc) { cerr << exc.what() << endl; exit(1); }

  return 0;
}
